modules/core/src/main/java/org/apache/ignite/spi/tracing/Scope.java

```diff
@@ -63,6 +70,20 @@ public short idx() {
         return idx;
     }
 
+    /**
+     * @return Coordinates without label.
+     */
+    @NotNull public TracingConfigurationCoordinates coordinates() {
+        return coordinates;
+    }
+
+    /**
+     * @return Coordinates with optional label.
+     */
+    @NotNull public TracingConfigurationCoordinates coordinates(@Nullable String label) {
```

Comment:
I would suggest removing this method and adding a new one to `TracingConfigurationCoordinates` as follows:

```
    /**
     * Creates a new tracing coordinates with the specified label. If the given {@code coordinates} already has the
     * specified label, the same instance is returned.
     *
     * @return Tracing coordinates with the specified label.
     */
    public static TracingConfigurationCoordinates of(TracingConfigurationCoordinates coordinates, String lb) {
        if (Objects.equals(coordinates.label(), lb))
            return coordinates;

        return new TracingConfigurationCoordinates.Builder(coordinates.scope()).withLabel(lb).build();
    }
```
and use it everywhere you need:
```
    TracingConfigurationCoordinates coordinates = TracingConfigurationCoordinates.of(scope.coordinates(), lb);
```

In that case, the code that creates new coordinates is a bit closer to the `TracingConfigurationCoordinates` itself. What do you think?